src/App.tsx

```diff
@@ -1,39 +1,81 @@
+/* eslint-disable no-console */
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock/Clock';
+
+interface State {
+  clockName: string;
+  hasClock: boolean;
+}
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends React.Component<{}, State> {
+  state: State = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
+
+  timerId = 0;
+
+  componentDidMount(): void {
+    this.timerId = window.setInterval(() => {
+      this.setState({
+        clockName: getRandomName(),
+      });
+    }, 3300);
+
+    document.addEventListener('contextmenu', (event) => {
+      event.preventDefault();
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+      this.hideClock();
+    });
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.addEventListener('click', () => {
+      this.showClock();
+    });
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  componentWillUnmount(): void {
+    window.clearInterval(this.timerId);
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+    document.removeEventListener('contextmenu', (event) => {
+      event.preventDefault();
 
-        {' time is '}
+      this.setState({
+        hasClock: false,
+      });
+    });
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+    document.removeEventListener('click', () => {
+      this.setState({
+        hasClock: true,
+      });
+    });
+  }
+
+  showClock() {
+    this.setState({
+      hasClock: true,
+    });
+  }
+
+  hideClock() {
```

Comment:
Event.preventDefault() should be in this handler

Reply:
done